rs/nns/sns-wasm/tests/add_wasm.rs

```diff
@@ -147,17 +145,19 @@ fn test_metadata_migration() {
         );
     }
 
-    // Check tha the metadata is not yet available.
+    // Check that the metadata is populated.
     {
         let response = get_wasm_metadata(&machine, sns_wasm_canister_id, &root_hash);
-        assert_eq!(
-            response,
-            GetWasmMetadataResponse {
-                result: Some(Result::Error(SnsWasmError {
-                    message: "get_wasm_metadata is not implemented yet.".to_string(),
-                })),
-            }
-        );
+        let GetWasmMetadataResponse {
+            result: Some(Result::Ok(Ok { sections })),
+        } = response
+        else {
+            panic!(
+                "Unexpected response from SnsW.get_wasm_metadata: {:?}",
+                response
+            );
+        };
```

Comment:
NCR

I am a fan of let else, but this is maybe too complicated. I think we should only use it when

1. everything up to `else {` fits on one line, and
2. there is only one layer of unwrapping.

It would look something like

```
let X(x) = x else {
    ...
};
```

Otherwise, I would consider falling back to match. E.g.

```suggestion
        let response = match response {  // Look at how simple this line is.
            GetWasmMetadataResponse { result } => result,
            otherwise => panic!(...),
        };
```

As hinted here ☝️ , I would even consider breaking out the various layers of unwrapping.

Another approach is to add some blank lines, and a comment:

```suggestion

        // Unwrap response.
        let GetWasmMetadataResponse {
            result: Some(Result::Ok(Ok { sections })),
        } = response
        else {
            panic!(
                "Unexpected response from SnsW.get_wasm_metadata: {:?}",
                response
            );
        };

```

Reply:
Thanks for caring about code readability; not many do.

The first suggestion is not an equivalent refactor, so I'm going with your second suggestion.